nrf_security: cracen: Add HMAC SHA256 to KMU key types.

ArekBalysNordic opened the pull request:
Added HMAC SHA256 as a new KMU key type.

NordicBuilder commented:
## CI Information

<sup>_To view the history of this post, clich the 'edited' button above_</sup>
Build number: 1
### Inputs:
#### Sources:
sdk-nrf: PR head: [d22a796287e641f3ca7abbbbf33d9f8431430eed](https://github.com/nrfconnect/sdk-nrf/pull/19582/commits/d22a796287e641f3ca7abbbbf33d9f8431430eed)
<details><summary> more details</summary>

sdk-nrf:
> PR head: [d22a796287e641f3ca7abbbbf33d9f8431430eed](https://github.com/nrfconnect/sdk-nrf/pull/19582/commits/d22a796287e641f3ca7abbbbf33d9f8431430eed)
> merge base: [d297557284fcddf77b2a1ef1edd2b77e8fd2975c](https://github.com/nrfconnect/sdk-nrf/commit/d297557284fcddf77b2a1ef1edd2b77e8fd2975c)
> target head (main): [d297557284fcddf77b2a1ef1edd2b77e8fd2975c](https://github.com/nrfconnect/sdk-nrf/commit/d297557284fcddf77b2a1ef1edd2b77e8fd2975c)
> [Diff](https://github.com/nrfconnect/sdk-nrf/compare/d297557284fcddf77b2a1ef1edd2b77e8fd2975c..d22a796287e641f3ca7abbbbf33d9f8431430eed)</details>

#### Github labels
| Enabled | Name | Description |
| -- | -- | -- |
|| ci-disabled | _Disable the ci execution_ | 
|| ci-all-test | _Run all of ci, no test spec filtering will be done_ | 
|| ci-force-downstream | _Force execution of downstream even if twister fails_ | 
|| ci-run-twister | _Force run twister_ | 
|| ci-run-zephyr-twister | _Force run zephyr twister_ | 

<details><summary> List of changed files detected by CI (2)</summary>

```
subsys
│  ├── nrf_security
│  │  ├── src
│  │  │  ├── drivers
│  │  │  │  ├── cracen
│  │  │  │  │  ├── cracenpsa
│  │  │  │  │  │  ├── src
│  │  │  │  │  │  │  ├── key_management.c
│  │  │  │  │  │  │  │ kmu.c
```
</details>

### Outputs:

#### Toolchain
Version: 
Build docker image: 

#### Test Spec & Results: :white_check_mark: Success; :x: Failure; :orange_circle: Queued; :yellow_circle: Progress; :white_medium_square: Skipped; :warning: Quarantine
- :orange_circle: Toolchain
- :orange_circle: Build twister
- :orange_circle: Integration tests
   - :orange_circle: test-fw-nrfconnect-chip
   - :orange_circle: test-fw-nrfconnect-nrf_crypto
   - :orange_circle: test-fw-nrfconnect-tfm
   - :orange_circle: test-sdk-find-my
   - :orange_circle: test-sdk-sidewalk
   - :orange_circle: test-sdk-dfu
   - :warning: test-fw-nrfconnect-nrf-iot_cloud

<details><summary>Disabled integration tests</summary>

- 
   - desktop52_verification
   - doc-internal
   - test_ble_nrf_config
   - test-fw-nrfconnect-apps
   - test-fw-nrfconnect-ble_mesh
   - test-fw-nrfconnect-ble_samples
   - test-fw-nrfconnect-boot
   - test-fw-nrfconnect-fem
   - test-fw-nrfconnect-nfc
   - test-fw-nrfconnect-nrf-iot_libmodem-nrf
   - test-fw-nrfconnect-nrf-iot_lwm2m
   - test-fw-nrfconnect-nrf-iot_mosh
   - test-fw-nrfconnect-nrf-iot_nrf_provisioning
   - test-fw-nrfconnect-nrf-iot_positioning
   - test-fw-nrfconnect-nrf-iot_samples
   - test-fw-nrfconnect-nrf-iot_serial_lte_modem
   - test-fw-nrfconnect-nrf-iot_thingy91
   - test-fw-nrfconnect-nrf-iot_zephyr_lwm2m
   - test-fw-nrfconnect-ps
   - test-fw-nrfconnect-rpc
   - test-fw-nrfconnect-rs
   - test-fw-nrfconnect-thread
   - test-fw-nrfconnect-zigbee
   - test-low-level
   - test-sdk-audio
   - test-sdk-mcuboot
   - test-sdk-pmic-samples
   - test-sdk-wifi
   - test-secdom-samples-public
</details>



*Note: This message is automatically posted and updated by the CI*

annwoj commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -68,7 +68,7 @@ Working with nRF54H Series
 =============================
 
-|no_changes_yet_note|
+* Added HMAC SHA-256 with a 128-bit key type to KMU :ref:`ug_nrf54l_crypto_kmu_supported_key_types`.
```
```suggestion
* Added HMAC SHA-256 with a 128-bit key type to KMU, as detailed in the :ref:`ug_nrf54l_crypto_kmu_supported_key_types` documentation section.
```

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='https://ncsdoc.z6.web.core.windows.net/PR-19582
/'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publish GitHub Action.*

ArekBalysNordic commented:
There are SUIT companion APP failures on Twister, not related to this PR. We need to wait for the fix there to rerun.

ArekBalysNordic commented:
Fixed a mistake, HMAC key must be pushed to RAM before using it by CRACEN. Added it to kmu.c. Also, there was no setting the opaque size for KMU HMAC key usage - added it to common.c
